v2/contribute/decisions/thirdparty/0001-showing-thirdparty-providers-list-dynamically.mdx

```diff
@@ -0,0 +1,106 @@
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Showing thirdparty provider list dynamically
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-07" created="2022-12-07" deciders={["rishabhpoddar", "sattvikc"]} proposedBy={["sattvikc"]} />
+
+## Context and Problem Statement
+
+### Context
+
+We would like to allow users to show the list of thirdparty providers dynamically. This means that the user should be able to add, update or delete providers without having to change the code. In the case of multi-tenancy, the user should be able to show different providers for different tenants.
+
+This is in the context of thirdparty recipe, however when we expand the multi-tenancy functionality to other recipes, we should be able to dynamically determine different login methods (emailpassword, thirdparty, etc) for each tenant.
+
+From the above context, we have 4 different use cases:
+1. Static Config and single tenant
+2. Static Config and multiple tenants
+3. Dynamic config and single tenant
+4. Dynamic config and multi tenant
+
+Clearly, for the use cases 3 & 4, we need to fetch the list of providers from the backend.
+
+### Use cases explained
+
+** 1. Static Config and single tenant **
+
+This is a very basic use case where the user has a single tenant and statically configured list of login methods and thirdparty providers.
+
+** 2. Static Config and multiple tenants **
+
+In this case, the user uses the same set of statically configured login methods and thirdparty providers across all tenants.
+
+** 3. Dynamic config and single tenant **
+
+In this scenario, although the user has a single tenant, he may need to add or remove thirdparty providers dynamically.
+
+** 4. Dynamic config and multi tenant **
+
+This would be a classic multi-tenant scenario, where each tenant has a different set of thirdparty providers enabled, and different credentials for them.
+
+### Problem Statement
+
+What would be the best way for the user to differentiate & configure the SDK for all the above use cases?
+
+## Considered Options
+
+* Show providers list dynamically when Multitenancy recipe is initialized
+* Always try to show providers list dynamically, fallback to statically configured list
+* Add `usesDynamicLoginMethods` config to the top level config
+
+## Decision Outcome
+
+Chosen option: **Add `usesDynamicLoginMethods` config to the top level config**, because
+
+* Clearly indicates that the dynamic providers list is being used
+* Avoids call to backend if the user does not intend to use dynamic providers list
+
+Also, `usesDynamicLoginMethods` is added as a toplevel config instead of thirdparty recipe, as this functionality will be expanded to other recipes in future.
```

Comment:
adding it to the top level and not recipe level is not cause "functionality will be expanded to other recipes" - we can even add this to each recipe. It's done to avoid several network calls. Please write the actual reason for things.

Reply:
Done